File: 이솔/프로그래머스/Level2/영어끝말잇기.py
def solution(n, words):
    answer=[0,0]
    arr=[]
    arr.append(words[0])
    for i in range(1,len(words)):
        pre_word=list(map(str,words[i-1]))
        word=list(map(str,words[i]))
        if words[i] in arr or pre_word[-1] !=word[0] :
            print(words[i],pre_word[-1],word[0])
            a=(i+1)%n
            b=((i+1)//n)+1
            if a==0:
                a=n
                b-=1
            
            
            answer=[a,b]
            break
            
        arr.append(words[i])
        
    return answer

File: 문민우/프로그래머스/level2/영어끝말잇기.java
package level2;

import java.util.Map;
import java.util.HashMap;

public class 영어끝말잇기 {
    public int[] solution(int n, String[] words) {

        int[] answer = new int[2];

        Map<String, Integer> map = new HashMap<>();

        for (int i = 0; i < words.length; i++) {

            if (i != 0) {

                String prev = words[i - 1];
                String cur = words[i];

                char last = prev.charAt(prev.length() - 1);
                char first = cur.charAt(0);

                if (map.containsKey(cur) || last != first) {

                    answer[0] = (i % n) + 1;
                    answer[1] = (i / n) + 1;

                    return answer;
                }
            }

            map.put(words[i], 1);
        }

        return answer;
    }
}
